### PR TITLE
[full-ci] thumbnails: add libvips based thumbnail generator

### PR DESCRIPTION
**Pull request by rhafer**
~~Needs: https://github.com/owncloud-ci/golang/pull/149~~

To improve performance (and to be able to support a wider range of images formats in the future)
the thumbnails service is now able to utilize libvips (https://www.libvips.org/) for generating thumbnails.
Enabling the use of libvips is implemented as a build-time option which is currently disabled for the
"bare-metal" build of the ocis binary and enabled for the docker image builds.

See README change for more details.

### CODE REVIEW

File: services/thumbnails/README.md [hunk 1]
```diff
@@ -89,3 +89,29 @@ To have more control over memory (and CPU) consumption the maximum number of con
...
 
 If a resource is shared using SecureView, the share reciever will get a 403 (forbidden) response when requesting a thumbnail. The requesting client needs to decide what to show and usually a placeholder thumbnail is used.
+
+## Using libvips for thumbnail generation
+
+To improve performance (and to support a wider range of images formats) the thumbnails service is able to utilize libvips (https://www.libvips.org/) for thumbnail generation. Support for libvips needs to be
+enabled at buildtime and has a couple of implications:
+
+*  With libvips support enabled it is not possible to create a statically linked ocis binary
+*  So the libvips shared libraries to be available at runtime in the same release that was used to build the ocis binary
```
Comment (butonic) [inline services/thumbnails/README.md:99]:
```suggestion
*  So the libvips shared libraries need to be available at runtime in the same release that was used to build the ocis binary
```

File: services/thumbnails/README.md [hunk 2]
```diff
@@ -89,3 +89,29 @@ To have more control over memory (and CPU) consumption the maximum number of con
 ## Thumbnails and SecureView
 
 If a resource is shared using SecureView, the share reciever will get a 403 (forbidden) response when requesting a thumbnail. The requesting client needs to decide what to show and usually a placeholder thumbnail is used.
+
+## Using libvips for thumbnail generation
+
+To improve performance (and to support a wider range of images formats) the thumbnails service is able to utilize libvips (https://www.libvips.org/) for thumbnail generation. Support for libvips needs to be
+enabled at buildtime and has a couple of implications:
```
Comment (mmattel) [inline services/thumbnails/README.md:96]:
```suggestion
## Using libvips for Thumbnail Generation

To improve performance and to support a wider range of images formats, the thumbnails service is able to utilize the [libvips library](https://www.libvips.org/) for thumbnail generation. Support for libvips needs to be
enabled at buildtime and has a couple of implications:
```

File: services/thumbnails/README.md [hunk 3]
```diff
@@ -89,3 +89,29 @@ To have more control over memory (and CPU) consumption the maximum number of con
...
+
+*  With libvips support enabled it is not possible to create a statically linked ocis binary
+*  So the libvips shared libraries need to be available at runtime in the same release that was used to build the ocis binary
+*  When using the ocis docker images the libvips shared libraries are included in the image
```
Comment (mmattel) [inline services/thumbnails/README.md:100]:
```suggestion
*  With libvips support enabled, it is not possible to create a statically linked ocis binary.
*  Therefore, the libvips shared libraries need to be available at runtime in the same release that was used to build the ocis binary.
*  When using the ocis docker images, the libvips shared libraries are included in the image and are correctly embedded.
```

File: services/thumbnails/README.md [hunk 4]
```diff
@@ -89,3 +89,29 @@ To have more control over memory (and CPU) consumption the maximum number of con
...
+
+Support of libvips is disabled by default. To enable it make sure libvips and its buildtime dependencies are install in your build environment. Then just set
+the `ENABLE_VIPS` variable on the `make` command:
```
Comment (mmattel) [inline services/thumbnails/README.md:103]:
```suggestion
Support of libvips is disabled by default. To enable it, make sure libvips and its buildtime dependencies are installed in your build environment.
Then you just need to set the `ENABLE_VIPS` variable on the `make` command:
```
Does one know how `buildtime dependencies are installed` works?
Should we add some text/links?

**Comment by rhafer** [comment]
The acceptance test are currently still using the native go thumbnailer. We might want to change that to test the libvips variant instead).

File: services/thumbnails/README.md (hunk 4 above)
Comment (rhafer) [reply services/thumbnails/README.md]: I tend to think that somebody trying to build ocis himself/herself should know how to install build dependencies.

**Comment by micbar** [comment]
I just compiled ocis on Mac (Intel)

## Steps

```
brew install vips pkg-config
export PKG_CONFIG_PATH="/usr/local/opt/libffi/lib/pkgconfig"
```

Than it compiles sucessfully and the thumbnails are generated.

**Review by micbar — APPROVED** [review]
Awesome! Tested it locally. I see no regressions. 🚀

**Comment by ScharfViktor** [comment]
> I just compiled ocis on Mac (Intel)

checked on mac (m1) - works fine 👍

**Comment by rhafer** [comment]
I just found an issue with rendering thumbnails for txt files. :man_facepalming: . Working on it ...

**Comment by ScharfViktor** [comment]
I tried to run e2e test which checks thumbnail and preview. 
for many preview requests we get 500 error

image


How to run: `BASE_URL_OCIS=localhost:9200 HEADLESS=false RETRY=0  pnpm test:e2e:cucumber tests/e2e/cucumber/features/shares/share.feature:79`

to enable e2e test in CI please rebase (e2e tests were disabled) https://github.com/owncloud/ocis/pull/10325

**Comment by rhafer** [comment]
> I tried to run e2e test which checks thumbnail and preview.
....
> to enable e2e test in CI please rebase (e2e tests were disabled) https://github.com/owncloud/ocis/pull/10325

@ScharfViktor Ok. I rebased. e2e test seem to have succeeded now.

**Comment by prohtex** [comment]
> I just compiled ocis on Mac (Intel)
> 
> ## Steps
> ```
> brew install vips pkg-config
> export PKG_CONFIG_PATH="/usr/local/opt/libffi/lib/pkgconfig"
> ```
> 
> Add `-tags enable_vips` to the go tool arguments of the debugger.
> 
> Then it compiles sucessfully and the thumbnails are generated.
> 
> First impression: 🚀

I have VIPS installed with MacPorts. Hope to see support enabled in a future release so I can test without building from source.

**Pull request merged**